src/shared/actions/transfers.js

```diff
@@ -427,124 +428,132 @@ export const makeTransaction = (seedStore, receiveAddress, value, message, accou
     let accountState = selectedAccountStateFactory(accountName)(getState());
     let transferInputs = [];
 
-    const withPreTransactionSecurityChecks = () => {
-        // Progressbar step => (Validating receive address)
-        dispatch(setNextStepAsActive());
+const withPreTransactionSecurityChecks = () => {
+    // Progressbar step => (Validating receive address)
+    dispatch(setNextStepAsActive());
 
-        // Make sure that the address a user is about to send to is not already spent.
-        return isAnyAddressSpent()([address])
-            .then((isSpent) => {
-                if (isSpent) {
-                    throw new Error(Errors.KEY_REUSE);
-                }
-
-                // Progressbar step => (Syncing account)
-                dispatch(setNextStepAsActive());
-
-                return syncAccount()(accountState, seedStore);
-            })
-            .then((newState) => {
-                // Assign latest account but do not update the local store yet.
-                // Only update the local store with updated account information after this transaction is successfully completed.
-                accountState = newState;
+    // Check the last trit for validity
+    if (!lastTritIsZero(address)) {
```

Comment:
We want this to be part of the promise chain (since there's a possibility that we throw an error here, we want it to throw and then go to the `catch` block and show the error message to the user), so I think the best approach would be similar to what we had before (but incorporate @laumair 's suggestions):
```js
return Promise.resolve(isLastTritZero(address)).then((lastTritIsZero) => {
  if(!lastTritIsZero) {
    throw new Error(Errors.INVALID_LAST_TRIT);
  }
  
  return `isAnyAddressSpent()([address])
  ...
```
The reason why we use these [promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises) is twofold: it makes error handling a bit easier, while also allowing us to use asynchronous functions without our code looking like [this](http://callbackhell.com/) :trollface: . So this is essentially what we're doing:

1. Call `isLastTritZero`, passing `address` as a parameter
2. Store whatever `isLastTritZero` returns in a variable called `lastTritIsZero` (in our case, this will be either true or false)
3. Is the value of `lastTritIsZero` ["falsy"](https://developer.mozilla.org/en-US/docs/Glossary/Falsy)? (so in our case, we're essentially checking if `lastTritIsZero === false`)
 a. i. If so, throw an error
 a. ii. When the error is thrown, we'll skip all the way to the `catch` block so we can figure out what to do with the error https://github.com/iotaledger/trinity-wallet/blob/252a1ed3de6f453beef52f6c510c376979990e3f/src/shared/actions/transfers.js#L726
  b. i. If not, then we'll continue by calling `isAnyAddressSpent`